### DISCUSSION TITLE
关于md文件导出的格式问题，On the format of md file export

### DISCUSSION THREAD

**Discussion by Xieading** [windingwind/zotero-better-notes#694] [Q&A] [answered]
当我用一个模板后，将笔记导出为md文件格式，把这个md文件用obsidian打开，会出现几个文档属性，文档属性内容是这篇文献的条目的几个元数据。我想问问大家有没有什么办法可以修改这里的文档属性，这样在导出这个笔记的时候自动将条目的元数据添加到文档属性中，这样可以让整个笔记非常简洁。
When I use a template, export the notes to an md file format, open the md file in obsidian, and several document attributes appear, the contents of which are several metadata of the entries in this document. I would like to ask if there is any way you can modify the document content here, so that the metadata of the entry will be automatically added to the document content when exporting this note, which will make this note very concise.
![屏幕截图 2023-09-15 171402](https://github.com/windingwind/zotero-better-notes/assets/110718481/d2e82437-a9f8-45fd-b7f9-6765c9819eff)

**Comment by Xieading** [comment]
我的意思是导出md文档在开头自带几个文档属性，类似YAML头。我想修改这一部分，让它在导出的时候带一些如作者、期刊之类的元数据。

**Comment by Xieading** [comment]
Issues #527  这位遇到了类似的问题

**Comment by windingwind** [comment, marked as answer]
使用`[ExportMDFileHeader]`模板添加自定义信息

**Reply by spencer-michaels** [reply]
I added some additional information, but the YAML still contains the extra values. For example, this is my `[ExportMDFileHeaderV2]` file: 
```
${{
  let yaml = {};
   yaml.title = noteItem.parentItem
    ? noteItem.parentItem.getField("title")
    : "";
  yaml.short = noteItem.parentItem
    ? noteItem.parentItem.getField("title-short")
    : "";
  yaml.tags = "";
  yaml.authors = noteItem.parentItem.getCreators().map((au) => au.firstName + " " + au.lastName).join("; ");
  yaml.year = noteItem.parentItem.getField("date") ? new Date(noteItem.parentItem.getField("date")).getFullYear() : '';
  yaml.date_created = new Date().toISOString().split('T')[0];
  yaml.type = ""
  return JSON.stringify(yaml);
}}$
```

Which returns: 
```
title: 'Chapter 8: The Living Cell'
short: ""
tags: ""
authors: 'N. A. Miliutin; George R. Collins; William Alex'
year: 1975
date_created: '2023-12-08'
type: ""
version: 29410
libraryID: 1
itemKey: XC63EHQU
```

How do I keep the last three values from being included?

**Reply by windingwind** [reply]
They're used by the plugin for markdown syncing. You cannot and should not remove them.

**Reply by XoYaPeng** [reply]
这是一个很不错的模板！！！

**Reply by charle36** [reply]
If you are using Obsidian, you can use the plugin Metadata Hider to hide the properties bothering you without removing them from the file.

**Comment by XoYaPeng** [comment]
@windingwind 大佬，我看了关于zotero的ExportMDFileHeaderV2中yaml区的好多讨论，现在有个问题想请教下：
`${await new Promise(async (r) => {
let header = {};
header.$tags = noteItem.parentItem.getTags().map((_t) => _t.tag);
header.noteLink = Zotero.BetterNotes.api.convert.note2link(noteItem);
header.tags = noteItem.parentItem.getTags().map((_t) => _t.tag);
r(JSON.stringify(header));
})}`
上面是我的标签模板，我在tags前面加“$”字符时，标签是可以自动更新到ob中，ob里显示的是“$tags”，但ob却不能识别出后面的标签了，这个有啥解决办法吗？这种标签的显示如下图1和2；另外不在tags前加"$"时，标签在ob里是可以正常显示的，但是却不能自动更新，如下图3的显示。
我想要实现的是，在zotero这边添加标签，可以在ob里自动更新，并且能正常显示标签，大佬能给支支招吗？
![image](https://github.com/windingwind/zotero-better-notes/assets/130196652/6130bfb4-c7e9-4df1-9282-04bc36141c49)

**Reply by windingwind** [reply]
仅`$`开头的自动字段由插件控制双向同步，允许在每次同步均由zotero侧的数据为准对字段进行覆盖更新。因此非`$`开头的用户字段不会根据zotero侧的数据自动更新，仅在首次同步时根据zotero侧的数据进行生成。至于obsidian方面的显示/读取问题，我不了解，无法帮助

**Reply by XoYaPeng** [reply]
大概了解了，谢谢大佬(^🙏^)

**Comment by XoYaPeng** [comment]
@windingwind 大佬，想请教下fileHeader的属性里，怎么添加下图中Zoterostyle功能中的“简记”字段呢？还有一个问题就是条目的链接属性代码怎么写呢？有时间大佬帮我看下，谢谢你！！
![image](https://github.com/windingwind/zotero-better-notes/assets/130196652/21dc72b4-b3fb-4540-aca0-a48364a1a54d)

**Reply by windingwind** [reply]
> 怎么添加下图中Zoterostyle功能中的“简记”字段呢？

请咨询对应插件作者如何获取该字段数据，我不熟悉

> 还有一个问题就是条目的链接属性代码怎么写呢

不太理解“链接属性”是什么？

**Reply by XoYaPeng** [reply]
大佬，链接属性，其实就是我想同步下条目链接🔗😄。

**Comment by dhwtb2009** [comment]
分享一个我的导出yaml格式
```
${{
  let header = {};
  header.created = noteItem.dateAdded;
  header.modified = noteItem.dateModified;
  header. 篇名 = noteItem.parentItem
    ? noteItem.parentItem.getField("title")
    : "";
header.作者 = noteItem.parentItem.getCreators().map((au) => au.firstName + " " + au.lastName).join("; ");
  header. 期刊 = (
    await Zotero.Collections.getCollectionsContainingItems([
      (noteItem.parentItem || noteItem).id,
    ])
  ).map((c) => c.name);
  header. 出版时间 = noteItem.parentItem
    ? noteItem.parentItem.getField("date")
    : "";
  header.citekey = noteItem.parentItem 
  ? noteItem.parentItem.getField("citationKey") 
  : "";
  return JSON.stringify(header);
}}$
```
